### PR TITLE
Support empty schema type

### PR DESCRIPTION
**Pull request by yacchi**
Some generators (e.g. swagger-codegen, go-swagger) are output to any type value for 'empty schema'.
Added 'empty schema' support into python-client, go-server, and ts-client in this pull-request.

Other changes include:

- Add generated code comment for go-server
Official format:
https://golang.org/cmd/go/#hdr-Generate_Go_files_by_processing_source 

- Support boolean argument in go-server
Add new template for 'type=boolean' request parameter in go-server.

And some small updates.

'empty schema' examples:
```yaml
definitions:
  # 'properties' does not exist
  # https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.9.3.2.1
  Ex1:
    type: object
  # empty schema object
  Ex2: {}
```

Sorry for large changeset😥

### CODE REVIEW

**Comment by mristin** [comment]
Hi @yacchi!
Thanks a lot for this large changeset! Please give me a couple of days to review it since I am a bit busy at the moment. (I'll try to have it reviewed till Friday, but can not promise that.)

**Comment by mristin** [comment]
Hi @yacchi,
Sorry, I'll be able to do the review only this Monday.

File: tests/cases/go_server/empty_schema/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+swagger: '2.0'
```
Comment (mristin) [inline tests/cases/go_server/empty_schema/swagger.yaml:1]: Is there a reason why you define many types in this test cases instead of just defining a single empty definition (that would be `meta`, correct?)? 

I wrote a suggestion (see at the bottom of this file) that is a bit more succinct and tests only for empty schemas. Please double-check that it was what you intended, I might have missed an important detail.

File: tests/cases/go_server/empty_schema/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        description: is a test string property.
+      capacity:
+        $ref: "#/definitions/Capacity"
+      metadata: {}
+      data:
+        type: object
+    required:
+      - product_id
+      - metadata
+      - data
```
Comment (mristin) [inline tests/cases/go_server/empty_schema/swagger.yaml:93]: (Comment B)

(I wrote this suggestion in editor.swagger.io, please do double-check that it works with the code and expresses what you originally wanted to test.)

```suggestion
swagger: '2.0'
info:
  title: Dummy Test API
  description: Test code generation.
  version: 1.0.0
schemes:
  - https
basePath: /
tags:
  - name: product
consumes:
  - application/json
produces:
  - application/json
paths:
  /test_endpoint:
    get:
      operationId: test_endpoint
      tags:
        - product
      description: test empty schema
      consumes:
        - application/json
      produces:
        - application/json
      parameters:
        - name: required_empty_parameter
          in: body
          schema:
            $ref: '#/definitions/EmptyParameter'
          required: true
      responses:
        200:
          description: a product object
          schema:
            $ref: '#/definitions/WithEmptyProperties'
        default:
          description: Unexpected error

definitions:
  EmptyParameter:
    description: defines an empty parameter.
    type: object

  WithEmptyProperties:
    description: is a product detail.
    type: object
    properties:
      required_empty_property: {}
      optional_empty_property:
        type: object
    required:
      - required_empty_property
```

File: swagger_to/go_server.py [hunk 3]
```diff
@@ -561,15 +573,16 @@ def _escaped_str(text: str) -> str:
 {% if fielddef.name not in typedef.required %}
 {% set json_tags = json_tags + ',omitempty' %}
 {% endif %}
-    {{ fielddef.name }} {{ field_type[fielddef] }} `json:{{ json_tags|escaped_str }}`{% if loop.last %}}{% endif %}
```
Comment (mristin) [inline swagger_to/go_server.py:564]: If I remember correctly, this was necessary if we want to handle nested objects in the future.

Was it only a cosmetics change or there was a deeper reason why you moved the curly brackets on the new line?

If it's only cosmetics: could you undo it in the pull request to keep it a bit easier to trace. After merging this pull request, I'll create an additional one to fix this issue. What do you think?

File: tests/cases/py_client/empty_schema/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
+swagger: '2.0'
```
Comment (mristin) [inline tests/cases/py_client/empty_schema/swagger.yaml:1]: Please synchronize with the suggested schema above, see Comment B.

**Comment by mristin** [comment]
P.S. would you mind rewriting commit messages to past tense to keep them consistent with the git log?

**Comment by yacchi** [comment]
Thanks for your review and helpful suggestions.

> Avoid introducing skip_type_conversion attribute?

I've got the same result as the type-based approach.
Removed that attribute in e809752.

> Make the unit test succinct to make it more explicit to the reader what is being tested.

Exactly. And the suggestions are correct.
Replace swagger definitions and a separate test for 'typed parameter tests' in b524a79 and a7470f7.

**Comment by mristin** [comment]
Thanks a lot, @yacchi ! Are you actively using swagger-to? Should we maybe wait for two weeks till the new minor version is released in case there are more patches, errors *etc*.?

**Comment by yacchi** [comment]
Thanks for merge this pull-req, @mristin!

Currently, I develop to 'file stream response support' for python client.
If you give me a couple of days, I hope to create a new pull-req for that changeset.

short sample:

If they write response schema type with 'file'

```yaml
...
operationId: file_stream
  responses:
    200:
      description: serves the file content.
      schema:
        type: file
...
```

Generated interface

```python
def file_stream(self) -> BinaryIO:
    ...
```

**Comment by mristin** [comment]
Great! No hurry -- let's wait then with the minor version.
